rust/pact_models/src/matchingrules.rs

```diff
@@ -1065,10 +1071,10 @@ impl MatchingRules {
     let category = category.into();
     match category {
       Category::BODY => self.rules_for_category(Category::BODY).map(|category| category.filter(|&(val, _)| {
-        calc_path_weight(val, path).0 > 0 && path_length(val) == path.len()
+        val.matches_path_exactly(path)
       })),
       Category::HEADER | Category::QUERY => self.rules_for_category(category.clone()).map(|category| category.filter(|&(val, _)| {
-        path.len() == 1 && path[0] == *val
+        path == &["$"] && val.is_root()
```

Comment:
This is the same as before, we want to check that the path contains the given header or query parameter name

Reply:
I see you have created a function for this: `first_field`